### PR TITLE
Verify well component

### PR DESCRIPTION
**Pull request by meissadia**
Part of SBL Project # 45

## Changes

- Add unit tests
- Add DS intro paragraph + source link
- Sync story content with DS, while making agreed upon improvements

## How to test this PR

1. yarn vitest Well

## Screenshots
Screenshot 2023-11-21 at 3 35 32 PM

### CODE REVIEW

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
Heyo! 2 small comments and 1 change requested. 👍

File: src/components/Well/Well.stories.tsx [hunk 1]
```diff
@@ -13,33 +13,12 @@ type Story = StoryObj<typeof meta>;
...
-      <a href='#' key='example1'>
-        Example link
-      </a>
+      <Link type='list' href='#' key='example1'>
+        Call-to-action link
+      </Link>
     ]
   }
-};
+};
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/components/Well/Well.stories.tsx:24]: I'm all for a linting rule that applies it as it's certainly not something I would want to watch for.  Candidly I didn't even know what was a "thing" 😬

File: src/components/Well/Well.tsx [hunk 2]
```diff
@@ -10,6 +10,11 @@ interface WellProperties {
   className?: string | undefined;
 }
 
+/**
+ * Wells are used to highlight specific information within a designated section of a page. This breaks up the flow of content on the page and helps to emphasize and set apart the content that is included. They are generally used on <a href="https://cfpb.github.io/design-system/pages/browse-pages">browse), <a href="https://cfpb.github.io/design-system/pages/learn-pages">learn), and <a href="https://cfpb.github.io/design-system/pages/document-detail-pages">document detail pages).
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/components/Well/Well.tsx]: Careless oversight on my part, thanks for being diligent!

File: src/components/Well/Well.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (billhimmelsbach) [reply src/components/Well/Well.stories.tsx:24]: It's super hard to even catch, so linters to the rescue! 🦸‍♂️

**Comment by meissadia** [comment]
@billhimmelsbach @natalia-fitzgerald I'd like to have a quick chat about this component when possible. I think that this implementation may be an incorrect interpretation of what constitutes a `Well`.  
 
From our `Complete user profile` prototype talks/explorations it seems that the `Well` is actually the grey box w/ border & padding that could house *any* content. 

The [DS example](https://cfpb.github.io/design-system/patterns/wells#well) demonstrates usage with a Text Introduction-like set of elements but I don't think that content is part of the `Well`.

**Comment by natalia-fitzgerald** [comment]
Here's my thinking (we can discuss more tomorrow): 
- Add the actual well to the DS components section (`o-well` in other words the gray box with padding)
- Add this well component (gray box with padding) to the the DSR
- Propose a name change to the well pattern in the CFPB DS. 
- Add the newly named well pattern thingy to our component library (o-well + heading + body/paragraph + link (optional)

The catch about what's referred to as the _well pattern_ is that this can be implemented in a number of ways. The heading can be an H3, H4 or H5. It can include or not include paragraph text and a call-to-action link. I'd have to check Wagtail to see all the customization that is possible but it's not a single thing in terms of implementation. 

As a UI designer I don't think of a well as a specific combination of elements. I think of a gray box with the padding. There is a standard implementation - which is the gray box with the heading, paragraph text, and sometimes a standalone link. But we also use the well organism (o-well) in the form context. For example: https://www.consumerfinance.gov/privacy/records-access/

There are other instances of gray boxes with a different padding that don't appear to use the o-well component. So maybe the o-well component is specifically the gray box with the 30px of padding. From a visual perspective what we really we have 2 types of boxes within our components - Box with 30px of padding, Box with 15px of padding. If we're pulling this out to the most simple pieces.

**Comment by natalia-fitzgerald** [comment]
@billhimmelsbach @meissadia 

Didn't we decide that these would be called: 

- Well (container)
  - No content, just the box with the fill, border, passing, responsive behavior
- Well (content)
  - Placeholder would be the heading (h3), paragraph, and stand-alone link (list link)
  
Like this?  
Screenshot 2023-11-21 at 7 17 55 AM

This is another example of where we currently use the well in the mock-ups:
- heading (h3), paragraph
 Screenshot 2023-11-21 at 7 19 03 AM
 
 Is there a specific place that we imagine that we will want to use the Well (container) or is that a nice to have in case it's useful for another component we are composing?

**Comment by billhimmelsbach** [comment]
So to sum up the changes to get this PR approved @meissadia:
- change the titles of the stories to Well (container) and Well (content)
- remove the content from the Well (container)
- change h2 to h3 in Well (content) example

**Comment by meissadia** [comment]
Updated to address feedback.  Screenshot in PR description has been updated. Tests updated.

**Review by billhimmelsbach — APPROVED** [review]
Looks good! 👍

**Pull request merged**